tests/ui/async-await/async-closures/overlapping-projs.rs

```diff
@@ -0,0 +1,27 @@
+//@ aux-build:block-on.rs
+//@ edition:2021
+//@ run-pass
+//@ check-run-results
+
+#![feature(async_closure)]
+
+extern crate block_on;
+
+async fn call_once(f: impl async FnOnce()) {
+    f().await;
+}
+
+async fn async_main() {
+    let x = &mut 0;
+    let y = &mut 0;
+    let c = async || {
```

Comment:
Before this last commit, we accidentally matched the upvar for `y` against `x` because they both "match" (they both have no projections). Assertions should make sure we're not doing that, lol.